plugins/module_utils/network/sonic/config/mirroring/mirroring.py

```diff
@@ -0,0 +1,479 @@
+
+        erspan_requests = self.get_modify_erspan_requests(command)
+        if erspan_requests:
+            requests.extend(erspan_requests)
+
+        return requests
+
+    def get_delete_mirroring_requests(self, command, is_delete_all=False):
+        requests = []
+
+        if not command:
+            return requests
+
+        if is_delete_all:
+            requests.append(self.get_delete_mirror_session_request())
+            return requests
+
+        span = command.get('span', [])
+        erspan = command.get('erspan', [])
+
+        for ms in span:
+            name = ms['name']
+            if len(ms) == 1:
+                requests.append(self.get_delete_mirror_session_request(name))
+                continue
+            if ms.get('source'):
+                requests.append(self.get_delete_mirror_session_request(name, 'src_port'))
+            if ms.get('direction'):
+                requests.append(self.get_delete_mirror_session_request(name, 'direction'))
+            if ms.get('dst_port'):
+                requests.append(self.get_delete_mirror_session_request(name, 'dst_port'))
+
+        for ms in erspan:
+            name = ms['name']
+            if len(ms) == 1:
+                requests.append(self.get_delete_mirror_session_request(name))
+                continue
+            if ms.get('src_ip'):
+                requests.append(self.get_delete_mirror_session_request(name, 'src_ip'))
+            if ms.get('source'):
+                requests.append(self.get_delete_mirror_session_request(name, 'src_port'))
+            if ms.get('direction'):
+                requests.append(self.get_delete_mirror_session_request(name, 'direction'))
+            if ms.get('dscp') is not None:
+                requests.append(self.get_delete_mirror_session_request(name, 'dscp'))
+            if ms.get('gre'):
+                requests.append(self.get_delete_mirror_session_request(name, 'gre_type'))
+            if ms.get('ttl') is not None:
+                requests.append(self.get_delete_mirror_session_request(name, 'ttl'))
+            if ms.get('queue') is not None:
+                requests.append(self.get_delete_mirror_session_request(name, 'queue'))
+            if ms.get('dst_ip'):
+                requests.append(self.get_delete_mirror_session_request(name, 'dst_ip'))
+
+        return requests
+
+    def validate_want(self, want):
+        if not want:
+            return
+
+        span = want.get('span', [])
+        erspan = want.get('erspan', [])
+
+        if span and erspan:
+            for ms in span:
+                name = ms['name']
+                in_erspan = next((ems for ems in erspan if name == ems['name']), None)
+                if in_erspan:
+                    err_msg = "Names of SPAN and ERSPAN mirror sessions should not be duplicated."
+                    self._module.fail_json(msg=err_msg, code=400)
+
+    def preprocess_want(self, want):
+        new_want = remove_empties(want)
```

Comment:
This "remove_empties" invocation is not needed because it was already done on line 143 in "set_config" when obtaining the "want" that is passed to "set_state" which, in turn, passes it to this function.